### ISSUE TITLE
[BUG] VM Network's `vlan` property now in UI as string, validator expects `int` 

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#6309]
**Describe the bug**
In Harvester `Version: v1.3-43ee3b8-head ` - when building a VM Network it throws a API / UI issue.

**To Reproduce**
1. Go to Networks -> VM Networks
1. (just for testing) build a VM network off of MGMT
1. input a VLAN ID
1. it will throw an error talking about the go struct, how it can't unmarshal the json of string to an integer

**Workaround**
Edit as yaml to take something like:
```
    {"cniVersion":"0.3.1","name":"base","type":"bridge","bridge":"extra-br","promiscMode":true,"vlan":"104","ipam":{}}
```
To:
```
    {"cniVersion":"0.3.1","name":"base","type":"bridge","bridge":"extra-br","promiscMode":true,"vlan":104,"ipam":{}}
```
- paying attention to the change of `"vlan":"104"` to -> `"vlan":104` - in order to please the validator

**Expected behavior**
UI not trying to ship payload as string, but rather, ship payload as integer

**Environment**
 - Harvester ISO version: v1.3-head (08/11/24)
 - Underlying Infrastructure: HP DL160 (bare-metal)

**Additional context**
![Screenshot from 2024-08-11 16-50-06](https://github.com/user-attachments/assets/de4094e6-484d-42b7-9d0a-32e2f8e9b1cb)
![Screenshot from 2024-08-11 16-49-38](https://github.com/user-attachments/assets/a5311432-5bcb-4846-86af-fef0c00f388b)

**Comment by FrankYang0529** [comment]
Thanks for reporting the issue. I also can reproduce this issue on master-head.

**Comment by bk201** [comment]
I can reproduce in 1.3-head, but can't reproduce this in v1.3.1.

**Comment by TachunLin** [comment]
Verified failed on the latest `v1.4-head` (**v1.4-4fe69de-head**) and master-head (**master-97b50dc-head**)
I also encounter the similar issue while creating the `vlan1` network on the `mgmt` interface 

It would block the creation and prompt the following error message
  ```
  admission webhook "mutator.harvester-system.harvester-network-webhook" denied the request: Internal error occurred: could not create nad default/vlan1 because json: cannot unmarshal string into Go struct field NetConf.vlan of type int
  ```
  ![image](https://github.com/user-attachments/assets/958fc60a-4985-413d-b1b8-0dc892f30294)

  ![image](https://github.com/user-attachments/assets/dce76f43-0cf7-46c8-b3c2-70378626c154)


Attached the support bundle for the reference
[supportbundle_3c2a0704-abe7-41eb-894b-041b497c5bfc_2024-08-12T09-17-19Z.zip](https://github.com/user-attachments/files/16579686/supportbundle_3c2a0704-abe7-41eb-894b-041b497c5bfc_2024-08-12T09-17-19Z.zip)

**Comment by irishgordo** [comment]
@a110605 thanks for the fix :+1: :smile: - this looks great and is fixed in v1.3.2-rc1 - I'll go ahead and close this out

![Screenshot from 2024-08-16 10-07-07](https://github.com/user-attachments/assets/7bae7331-147c-42e7-b9d2-e007d9481243)
